src/rpc/request.cpp

```diff
@@ -130,20 +130,21 @@ void DeleteAuthCookie()
     }
 }
 
-std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue &in, size_t num)
+std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue& in)
 {
     if (!in.isArray()) {
         throw std::runtime_error("Batch must be an array");
     }
+    const size_t num {in.size()};
     std::vector<UniValue> batch(num);
-    for (size_t i=0; i<in.size(); ++i) {
+    for (size_t i=0; i<num; ++i) {
```

Comment:
c2d63a8c05e14cdfdaae2ebca7a6b38907c31f1c

```diff
     const size_t num {in.size()};
     std::vector<UniValue> batch(num);
-    for (size_t i=0; i<num; ++i) {
-        const UniValue &rec = in[i];
+    for (const UniValue &rec : in.getValues()) {
         if (!rec.isObject()) {
             throw std::runtime_error("Batch member must be an object");
```

Reply:
That's better -- thanks! Done.